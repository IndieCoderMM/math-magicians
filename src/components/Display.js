import React, { Component } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line
export default class Display extends Component {
  render() {
    const { operation, next, total } = this.props;
    return (
      <div className="output-screen">
        <div className="formula">
          {operation}
          {next}
        </div>
        <div className="output-text">{total || 0}</div>
      </div>
    );
  }
}

Display.defaultProps = {
  total: '',
  next: '',
  operation: '',
};

Display.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
  operation: PropTypes.string,
};
