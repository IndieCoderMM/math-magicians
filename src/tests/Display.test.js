import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Display from '../components/Display';

describe('Display Component', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display total value', async () => {
    // Arrange
    const total = '19999';
    render(<Display total={total} />);
    // Act
    const result = await screen.findByText(total);
    // Assert
    expect(result).toBeTruthy();
  });
});
