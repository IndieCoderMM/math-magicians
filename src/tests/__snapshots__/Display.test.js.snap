// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component should render correctly 1`] = `
<div
  className="output-screen"
>
  <div
    className="formula"
  />
  <div
    className="output-text"
  >
    0
  </div>
</div>
`;
